core/src/Kokkos_Serial.hpp

```diff
@@ -573,13 +570,14 @@ class ParallelReduce<FunctorType, Kokkos::RangePolicy<Traits...>, ReducerType,
             : pointer_type(
                   internal_instance->m_thread_team_data.pool_reduce_local());
 
-    reference_type update =
-        ValueInit::init(ReducerConditional::select(m_functor, m_reducer), ptr);
+    typename Analysis::template Reducer<> final_reducer(
```

Comment:
This syntax is really awkward.
Did you consider dropping the template parameter
https://github.com/kokkos/kokkos/blob/30fdaaad2e86ae952eb22160032b3da022e6c7e5/core/src/impl/Kokkos_FunctorAnalysis.hpp#L763
It is only used to define the public member `memory_space`
https://github.com/kokkos/kokkos/blob/30fdaaad2e86ae952eb22160032b3da022e6c7e5/core/src/impl/Kokkos_FunctorAnalysis.hpp#L785
which is not required by the reducer concept as far as I can tell.

Reply:
Sure, I can do that. 
Note that I don't intend this class to precisely match the Reducer concept, in particular, since this is the place where all member functions just take a pointer instead of differentiating between array and scala reductions taking pointers and references respectively. Also, `init` returning something is a convenience feature that comes in handy in a lot of backend implementations.